memory_model/src/guest_memory.rs

```diff
@@ -154,6 +268,24 @@ impl GuestMemory {
         Ok(())
     }
 
+    /// perform specified action on himem regions
+    pub fn with_himem_regions<F, E>(&self, cb: F) -> result::Result<(), E>
+    where
+        F: Fn(usize, GuestAddress, usize, usize) -> result::Result<(), E>,
+    {
+        for (index, region) in self.regions.iter().enumerate() {
```

Comment:
a slightly more idiomatic way to do this would be to filter the collection before iterating, so something like:

```rust
for (index, region) in self.regions.iter().enumerate().filter(|i, _| i > 0) {
    cb(...)?;
}
```

or in this case, since you're using all but the first element:

```rust
for (index, region) in self.regions.iter().skip(1).enumerate() {
    cb(...)?;
}
```